### DISCUSSION TITLE
Introduce new type of context

### DISCUSSION THREAD

**Discussion by DhSufi** [elgatosf/elgato#65] [Ideas]
I have been playing with the SDK and just noticed that once you have the context of an instance's action, then you can send the Events you want to that instance regardless it is your own plugin or third party plugin.

It would be great to have 2 contexts for instance's actions: one public and one private. The private one would be the same we have now. The public one would allow us to perform limited Events requets to any plugin (for example setImage or setTitle, allowed and getGlobalSettings or sendToPropertyInspector forbiden). This way a lot of people asking for changing images of other plugins/keys would be able to do it with a low security risk.

Would be that possible?
Thanks for your time and consideration
Regads

**Comment by GeekyEggo** [comment]
Hey @DhSufi, thanks for taking the time to provide us with feedback.

> Once you have the context of an instance's action, then you can send the Events you want to that instance regardless it is your own plugin or third party plugin.

Whilst it is possible to send a command to Stream Deck for any action context, as you have experienced, there are safe-guards in place to prevent cross-contamination of requests from plugins that don't "own" the action associated with said context. These safe-guards serve as a security measure to prevent sensitive information being leaked to other plugins, or a plugin acting maliciously.

At the moment we have no plans to incorporate cross-plugin communication as part of the SDK. Should two or more plugins wish to communicate with one another it would be required they establish a independent channel of communication, e.g. WebSocket connection.

**Comment by DhSufi** [comment]
Thank you so much for your informative answer, I really appreciate it.
Regards.
